packages/rpc/src/batchQuery.ts

```diff
@@ -13,82 +13,68 @@ export const keyToPath = (key: string) => {
   ];
 };
 
-export const batchVstorageQuery = (
+export const batchVstorageQuery = async (
   node: string,
   unmarshal: FromCapData<string>,
   paths: [AgoricChainStoragePathKind, string][],
 ) => {
-  const options = {
-    method: 'POST',
-    body: JSON.stringify(
-      paths.map((path, index) => ({
-        jsonrpc: '2.0',
-        id: index,
-        method: 'abci_query',
-        params: { path: `/custom/vstorage/${path[0]}/${path[1]}` },
-      })),
-    ),
-  };
+  const urls = paths.map(
+    path => new URL(`${node}/agoric/vstorage/${path[0]}/${path[1]}`).href,
+  );
+  const requests = urls.map(url => fetch(url));
 
-  return fetch(node, options)
-    .then(res => res.json())
-    .then(res =>
-      Object.fromEntries(
-        (Array.isArray(res) ? res : [res]).map(entry => {
-          const { id: index } = entry;
-          if (entry.result.response.code) {
-            return [
-              pathToKey(paths[index]),
-              { error: entry.result.response.log },
-            ];
-          }
-
-          if (!entry.result.response.value) {
-            return [
-              pathToKey(paths[index]),
-              {
-                error: `Cannot parse value of response for path [${
-                  paths[index]
-                }]: ${JSON.stringify(entry)}`,
-              },
-            ];
-          }
+  return Promise.all(requests)
+    .then(responseDatas => {
+      return Promise.all(responseDatas.map(res => res.json()));
+    })
+    .then(responses =>
+      responses.map((res, index) => {
```

Comment:
this handler is quite complex at this depth.

consider factoring it out into a named function and adding a unit test for it.

Reply:
The logic could be pulled out into some other functions, but the code-paths are well tested, and I prefer having the unit tests be more encompassing of the entire logic with chainStorageWatcher.ts, I'll just leave as-is for now